R/report.R

```diff
@@ -79,9 +79,17 @@ Report <- R6Class( #nolint: object_name_linter
       self$print(success, warning, error)
       sink()
     },
-    save_results = function(file_name, method = write.csv, ...) {
-      self$get_validations(unnest = TRUE) %>%
-        write.csv(file = file_name)
+    save_results = function(file_name = "results.csv", method = utils::write.csv, ...) {
+      func_args <- names(formals(method))
+      if (!any(func_args == "...") && !("file" %in% func_args))
```

Comment:
This is a very limiting and weak approach. Moreover, it is not documented. It is ok to ask a user to provide a save method with some specified parameters but:
1. Why do you expect that `...` with `file` parameter?
2. Why you expect that the object that should be saved is going to be the first or the second parameter?
None of this is checked. And with this approach, you give no suggestion on how to deal with methods that have different parameters, e.g. `path` instead of `file`.

Suggested approach:
- remove the parameters evaluation part
- add documentation that will inform a user what are the required parameters for the method (follow the convention from `utils` and `readr` with `x` and `file`)
- add examples of how to deal with methods with different parameter names - create a wrapper that will change the names
- make sure that all arguments are passed to the method so instead of using pipe, call `method(x = , file = )`
- add information about the required parameters to the error message that is emitted if the writing method fails
- everything needs to be tested